Adding option to order by DVD ordering

smeggingsmegger opened the pull request:
This is an important option for TV Shows to allow for returning the results in DVD order instead of Airdate Order. Why is this important? Because many people are ripping their TV shows and using SickBeard to manage them and automatically name and process them.

The change here allows you to add shows like "Batman: The Animated Series" and "Firefly" where the network screwed up the air ordering but fixed it in a subsequent DVD release.

This PR requires a library update: https://github.com/midgetspy/Sick-Beard/pull/743

smeggingsmegger committed:
Adding DVD ordering to Sick-Beard

I originally had a good commit messsage here but months of this PR
becoming stagnant and repeatedly rebasing have removed it.

It adds the option to order by DVD using the TVDB API. This fixes a
whole bunch of shows like Firefly, Futurama, Batman The Animated Series
and many more.

thezoggy commented:
i do believe you could just do `if self.dvdorder:` instead of `if self.dvdorder != 0:`

thezoggy commented:
you should submit the dvd order support to the tvdb_api rather than we fork their code. https://github.com/dbr/tvdb_api/

smeggingsmegger commented:
@thezoggy I will change it to `if self.dvdorder`

I didn't realize that lib/tvdb_api/tvdb_api.py was a separate project. I will submit a PR to them. Is there a reason that library isn't just a pip requirement?

smeggingsmegger commented:
@thezoggy Thank you for the comments. I've sent a PR to tvdb_api and I will remove and rebase my changes to tvdb_api.py inside of this PR as soon as they accept it. I am leaving them in for now in case people want to test my changes easily.

thezoggy commented:
Lessen the need for users to install.. Plus SB core depends so much on it. Wile the code looks fine, dbr will make sure and tweak if needed. You may want to add a test for your code (firefly)

-----Original Message-----
From: "Scott Blevins" notifications@github.com
Sent: ‎9/‎28/‎2013 11:04 AM
To: "midgetspy/Sick-Beard" Sick-Beard@noreply.github.com
Cc: "thezoggy" thezoggy@gmail.com
Subject: Re: [Sick-Beard] Adding option to order by DVD ordering (#742)

@thezoggy I will change it to `if self.dvdorder`

I didn't realize that lib/tvdb_api/tvdb_api.py was a separate project. I will submit a PR to them. Is there a reason that library isn't just a pip requirement? 

---

Reply to this email directly or view it on GitHub:
https://github.com/midgetspy/Sick-Beard/pull/742#issuecomment-25300747

smeggingsmegger commented:
@thezoggy that makes sense. Thank you.

I tried running the existing test suite and I got two failures and an error. I tried again in master but it seems the tests are broken at the moment.

If I had my way, I would use HTTPretty (https://github.com/gabrielfalcao/HTTPretty) to test this change with an integration test. That would add quite a bit of coverage for existing Sick-Beard code because it would use all the existing code and only override the call to thetvdb at the socket level. It would still be quick and deterministic since we would be defining the return value for the call and it wouldn't actually use the network.

I've done similar tests at my job and I am currently working on a test suite for the Paython library that uses this method.

The only reason I didn't go crazy and write that test was because I wasn't sure if my PR would be rejected for doing so and I didn't want to do all that work for nothing. If you gave me the go-ahead, I would work on an integration test suite for the tvdb calls in Sick-Beard using HTTPretty.

I'd also be happy to take a crack at fixing the existing tests so that we could set up some CI for this awesome project.

Let me know what you think. Thanks!

smeggingsmegger commented:
@thezoggy Thanks for the tip. I added a test for tvdb_api in my PR to them. I used Firefly and Batman The Animated Series and it is passing with flying colors. Hopefully they will merge it soon.

smeggingsmegger commented:
The tvdb_api merged my PR today. How should I handle this PR? Should I check out the new master for that project and blast it into the lib directory since my PR requires the updated library code?

thezoggy commented:
Yes, make a new pr off the updated lib

-----Original Message-----
From: "Scott Blevins" notifications@github.com
Sent: ‎10/‎1/‎2013 1:45 PM
To: "midgetspy/Sick-Beard" Sick-Beard@noreply.github.com
Cc: "thezoggy" thezoggy@gmail.com
Subject: Re: [Sick-Beard] Adding option to order by DVD ordering (#742)

The tvdb_api merged my PR today. How should I handle this PR? Should I check out the new master for that project and blast it into the lib directory?

---

Reply to this email directly or view it on GitHub:
https://github.com/midgetspy/Sick-Beard/pull/742#issuecomment-25476753

smeggingsmegger commented:
@thezoggy I've sent a library update PR in #743. Let me know if you need anything else.

smeggingsmegger commented:
@thezoggy is there anything wrong with this PR? I did the things you suggested. I am not sure what else I need to do to get this merged. Any suggestions would be appreciated.

thezoggy commented:
as this code requires another pr to be pulled, this is on hold until that pr is reviewed. just need to be patient

lawrencecurtis commented:
With the API updated, can this now be merged?

thezoggy commented:
can you rebase this off development first, looks like you still have tvdb_api changes in it

smeggingsmegger commented:
@thezoggy I rebased this off of development in case you wanted to merge it for @lawrencecurtis.

smeggingsmegger commented:
The tvdb_api merged my PR quite some time ago so the changes inside of lib/tvdb_api/tvdb_api.py reflect the changes that I added into their library.

thezoggy commented:
@smeggingsmegger I see some left over merge conflicts..

```
+<<<<<<< HEAD

 MIN_DB_VERSION = 9  # oldest db version we support migrating from

 MAX_DB_VERSION = 14

+||||||| merged common ancestors

+MAX_DB_VERSION = 12

+=======

+MAX_DB_VERSION = 13

+>>>>>>> Fixing conflicts.

```

smeggingsmegger commented:
@thezoggy Sorry about that. I was distracted and in a hurry while I was doing that. :-/ It's fixed now.

lawrencecurtis commented:
Can this now be merged in? I have about 5 TV shows all suffering from messed up ordering which this would fix

smeggingsmegger commented:
:+1: I'd love to see this merged in. I updated an external library and the changes to it were brought into Sick-Beard 5 months ago so we should be clear to merge now. I checked over my code and I don't see any outstanding issues.

@thezoggy What do you think?

lawrencecurtis commented:
I think it will need a few tweaks in the sickbeard/databases/mainDB.py as I'm getting an error... AddDvdOrderOption(UpgradeHistoryForGenericProviders) i changed that too AddDvdOrderOption(AddLastUpdateTVDB): and updated the db versions etc quickly to get it to run.  

So i think you might need to do another rebase from development and tweak that and then all they need is to hit merge

smeggingsmegger commented:
@lawrencecurtis You might want to check out my branch and make sure it is still working. I rebased and fixed some conflicts but it looks like metadata/xbmc.py has been completely rewritten and I didn't have time to see if my branch still works with it.

thezoggy commented:
should squash and rebase. force push to rewrite history to make it clean

smeggingsmegger commented:
I've rebased and squashed.

smeggingsmegger commented:
@thezoggy what do you think?

smeggingsmegger commented:
@thezoggy pinging this again.

smeggingsmegger commented:
Rebased off of upstream/development again so that it is ready to merge whenever you want.

thezoggy commented:
ty

smeggingsmegger commented:
Closing this PR as I'm using Sick Rage these days. Best of luck to your project.
